Development/nmos/sdp_utils.cpp

```diff
@@ -723,6 +723,44 @@ namespace nmos
             );
         }
 
+        // insert extmap if specified
+        for (const auto& extmap_entry : sdp_params.extmap)
+        {
+            // a=extmap:<value>["/"<direction>] <URI> <extensionattributes>
+            // See https://www.rfc-editor.org/rfc/rfc5285#section-5
+            web::json::push_back(
+                session_attributes, value_of({
+                    { sdp::fields::name, sdp::attributes::extmap },
+                    { sdp::fields::value, value_of({
+                        { sdp::fields::local_id, extmap_entry.local_id },
+                        { extmap_entry.direction != sdp::direction{} ? sdp::fields::direction.key : U(""), extmap_entry.direction.name },
+                        { sdp::fields::uri, extmap_entry.uri },
+                        { !extmap_entry.ext_attributes.empty() ? sdp::fields::extensionattributes.key : U(""), extmap_entry.ext_attributes },
+                    }, keep_order) }
+                }, keep_order)
+            );
+        }
+
+        // insert hkep if specified
+        for (const auto& hkep_attr : sdp_params.hkep)
+        {
+            // a=hkep:<port> <nettype> <addrtype> <unicast-address> <node-id> <port-id>
+            // See VSF TR-10-5 Section 10
+            web::json::push_back(
+                session_attributes, value_of({
+                    { sdp::fields::name, sdp::attributes::hkep },
+                    { sdp::fields::value, value_of({
+                        { sdp::fields::port, hkep_attr.port },
+                        { sdp::fields::network_type, sdp::network_types::internet.name },
+                        { sdp::fields::address_type, details::get_address_type_multicast(origin_address).first.name },
+                        { sdp::fields::unicast_address, origin_address },
```

Comment:
Is the logic of `nmos::details::get_origin_address()` always appropriate for `hkep`?

Reply:
Maybe, it's better to allow apps explicitly set it, you're right.